### PR TITLE
Led rewrite

### PR DESCRIPTION
**Pull request by HenryLi-0**
changes:
- no led enum, replaced with commands
- same classic led colors/patterns, the ones you know and love!

### CODE REVIEW

File: src/main/java/org/sciborgs1155/robot/Robot.java [hunk 1]
```diff
@@ -181,15 +180,13 @@ private void configureBindings() {
 
     drive.setDefaultCommand(drive.drive(x, y, omega));
 
-    led.setDefaultCommand(led.setLEDTheme(LEDTheme.ALLIANCE));
+    led.setDefaultCommand(led.alliance());
```
Comment (kishan243) [inline src/main/java/org/sciborgs1155/robot/Robot.java:183]: The default command should be a color when there is no note in the robot . You can do this for when the robot is disabled instead.

File: src/main/java/org/sciborgs1155/robot/led/LedStrip.java [hunk 2]
```diff
@@ -10,161 +10,137 @@
...
-          return new Color(
-              (int) ((-Math.sin(theta) + 1) * scalar),
-              (int) ((Math.sin(theta) + 1) * scalar),
-              (int) ((Math.cos(theta) + 1) * scalar));
+  private static Color[] raindrop = new Color[LED_LENGTH];
+
+  public Command raindrop() {
+    return run(
+        () -> {
+          if (Math.round(Math.random()) == 0) {
```
Comment (Yxhej) [inline src/main/java/org/sciborgs1155/robot/led/LedStrip.java]: This line of code can be easily simplified by testing if `Math.random() < 0.5. `

File: src/main/java/org/sciborgs1155/robot/led/LedStrip.java [hunk 3]
```diff
@@ -10,161 +10,137 @@
...
+    String[] ledBufferData = new String[ledBuffer.getLength()];
+    for (int i = 0; i < ledBuffer.getLength(); i++) {
+      ledBufferData[i] = ledBuffer.getLED(i).toHexString();
     }
+    return ("[" + String.join(",", ledBufferData) + "]");
+  }
 
-    for (int i = raindrop.length - 1; i > 0; i -= 1) {
-      raindrop[i] = raindrop[i - 1];
+  public static AddressableLEDBuffer genBuffer(Function<Integer, Color> f) {
```
Comment (Yxhej) [inline src/main/java/org/sciborgs1155/robot/led/LedStrip.java:131]: Documentation check 🔥🔥🔥

**Review by Yxhej — CHANGES_REQUESTED** [review]

File: src/main/java/org/sciborgs1155/robot/led/LedStrip.java (hunk 3 above)
Comment (Yxhej) [reply src/main/java/org/sciborgs1155/robot/led/LedStrip.java:131]: Cool change (honestly did not know that was a thing), but I meant to add a javadoc explaining what exactly it takes in and what it does. To unfamiliar eyes, the usage of this (and buffers in general) is unclear. Be as concise as possible.

**Review by Yxhej — APPROVED** [review]

**Pull request merged**